subsys/net/ip/net_if.c

```diff
@@ -261,6 +261,9 @@ static bool net_if_tx(struct net_if *iface, struct net_pkt *pkt)
 		net_if_tx_lock(iface);
 		status = net_if_l2(iface)->send(iface, pkt);
 		net_if_tx_unlock(iface);
+		if (status < 0) {
+			NET_WARN("iface %p pkt %p send failure status %d", iface, pkt, status);
```

Comment:
I suggest printing interface index instead of pointer, much easier to track things if there are multiple network interfaces.
So
```
NET_WARN("iface %d pkt %p send failure status %d", net_if_get_by_iface(iface), pkt, status);
```

Reply:
This file seems to be using a mix of both versions, should it be updated everywhere to `net_if_get_by_iface`?

Reply:
> This file seems to be using a mix of both versions, should it be updated everywhere to `net_if_get_by_iface`?

Sure, no objection to that. Of course in a separate commit.

Reply:
done